### PR TITLE
add speedlimitunit and speedlimitsign to legstep

### PR DESCRIPTION
**Pull request by abhishek1508**
As per speed limit requirements from the Directions, the API response needs to show the `speedLimitSign` and `speedLimitUnit` as properties in the `LegStep` object.

<details>
  <summary>Example response</summary>
  
  ```json
    "steps": [
             {
                 "intersections": [],
                 "driving_side": "right",
                 "geometry": "i`q`gAvrzohFTfD~@zNz@tM",
                 "mode": "driving",
                 "maneuver": {},
                 "weight": 32.6,
                 "duration": 8,
                 "name": "Fell Street",
                 "distance": 51,
                 "speedLimitSign": "mutcd",
                 "speedLimitUnit": "mph",
                 "voiceInstructions": [],
                 "bannerInstructions": []
             }
     ]
  ```
</details>

### CODE REVIEW

**Review by Guardiola31337 — COMMENTED** [review]
Did we test that Directions requests return speed limit signs and units and if they're parsed properly when using the `DirectionsService`? It'd be also great to test with a `SNAPSHOT` from the Navigation SDK and if it works release a beta so we can integrate it.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/LegStep.java [hunk 1]
```diff
@@ -20,6 +23,27 @@
...
+  /**
+   * Speed limit sign.
+   */
+  @Retention(RetentionPolicy.SOURCE)
+  @StringDef({
+    MUTCD,
+    VIENNA
+  })
+  public @interface SpeedLimitSign {
+  }
```
[the remark this answers is not in the document]
Comment (abhishek1508) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/LegStep.java:45]: I can do this, but I saw that `TypeDef` in `Mapbox-Java` are included in the class file who needs it. So just didn't want to break the pattern.
cc @Guardiola31337

**Comment by abhishek1508** [comment]
> Did we test that Directions requests return speed limit signs and units and if they're parsed properly when using the DirectionsService?

How do you test this? cc @Guardiola31337

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/LegStep.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Guardiola31337) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/LegStep.java:45]: Ohh, if that's the case, yeah let's leave it as is. I thought that we were actually creating a separate file in these cases. I'm curious, what class did you take as an example?

**Comment by Guardiola31337** [comment]
@abhishek1508 
> How do you test this?

Run a request that you know beforehand is going to return speed limit signs and units. Then, you can modify temporarily the `circle.yml` pointing to your branch https://github.com/mapbox/mapbox-java/blob/master/circle.yml#L81 so a `SNAPSHOT` is generated and you can integrate it in the Navigation SDK and test the client integration before merging. We should also add a MockWebServer test. Check https://github.com/mapbox/mapbox-java/pull/1207 when you have a chance, we went to that process (it's a similar change as the one we're adding here).

**Comment by abhishek1508** [comment]
Using the snapshot released I can see `speedLimitSign` and `speedLimitUnit` in the SDK

Screen Shot 2020-10-22 at 10 48 45 AM

cc @Guardiola31337

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/SpeedLimitUnit.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.mapbox.api.directions.v5.models;
+
+import androidx.annotation.StringDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class SpeedLimitUnit {
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/SpeedLimitUnit.java:8]:
```suggestion
public class SpeedLimit {
```

**Review by Guardiola31337 — APPROVED** [review]
Minor suggestion. Other than that looks good 👍 

Thanks for addressing the feedback @abhishek1508

**Comment by Guardiola31337** [comment]
There's also CI failing https://app.circleci.com/pipelines/github/mapbox/mapbox-java/244/workflows/73eed8f8-43b6-4c2a-a521-f5f50e39069d/jobs/4557

```
* What went wrong:
Execution failed for task ':services-directions-models:checkstyleMain'.
> Checkstyle rule violations were found. See the report at: file:///root/code/services-directions-models/build/reports/checkstyle/main.html
  Checkstyle files with violations: 11
  Checkstyle violations by severity: [warning:19, error:5]
```

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**